webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -2545,6 +2453,331 @@ class SearchResponderV2 extends ResponderWithStandoffV2 {
+                            val dependentResIriOption: Option[IRI] = resultRow.rowMap.get(dependentResVar.variableName)
+
+                            dependentResIriOption match {
+                                case Some(depResIri: IRI) =>
+
+                                    // IRIs are concatenated by GROUP_CONCAT using a separator, split them
+                                    depResIri.split(transformer.groupConcatSeparator).toSeq
+
+                                case None => Set.empty[IRI] // no Iri present since variable was inside aan OPTIONAL or UNION
+                            }
+
+                    }
+
+                    acc + (mainResIri -> dependentResIris)
+            }
+
+            DependentResourcesPerMainResource(new ErrorHandlingMap(dependentResourcesPerMainRes, { key => throw GravsearchException(s"main resource not found: $key") }))
+        }
+
+        /**
+          * Collects object variables and their values per main resource from the results returned by the prequery.
+          * Value objects variables and their Iris are grouped by main resource.
+          *
+          * @param prequeryResponse the results returned by the prequery.
+          * @param transformer      the transformer that was used to turn the Gravsearch query into the prequery.
+          * @param mainResourceVar  the variable representing the main resource.
+          * @return [[ValueObjectVariablesAndValueObjectIris]].
+          */
+        def getValueObjectVarsAndIrisPerMainResource(prequeryResponse: SparqlSelectResponse,
+                                                     transformer: NonTriplestoreSpecificConstructToSelectTransformer,
+                                                     mainResourceVar: QueryVariable): ValueObjectVariablesAndValueObjectIris = {
+
+            // value objects variables present in the prequery's WHERE clause
+            val valueObjectVariablesConcat = transformer.getValueObjectVarsGroupConcat
+
+            val valueObjVarsAndIris: Map[IRI, Map[QueryVariable, Set[IRI]]] = prequeryResponse.results.bindings.foldLeft(Map.empty[IRI, Map[QueryVariable, Set[IRI]]]) {
+                (acc: Map[IRI, Map[QueryVariable, Set[IRI]]], resultRow: VariableResultsRow) =>
+
+                    // the main resource's Iri
+                    val mainResIri: String = resultRow.rowMap(mainResourceVar.variableName)
+
+                    // the the variables representing value objects and their Iris
+                    val valueObjVarToIris: Map[QueryVariable, Set[IRI]] = valueObjectVariablesConcat.map {
+                        valueObjVarConcat: QueryVariable =>
+
+                            // check if key exists: the variable representing value objects
+                            // could be contained in an OPTIONAL or a UNION and be unbound
+                            // It would be suppressed by `TriplestoreMessages` in that case.
```

Comment:
What is `TriplestoreMessages`? I can't find this class.

Reply:
done in 
https://github.com/dhlab-basel/Knora/pull/1028/commits/5eb44a03e657b929b4c247009d0d1d7bfc94ca57